orb-ui/cone/examples/cone-simulation.rs

```diff
@@ -0,0 +1,188 @@
+                                        for pixel in pixels.iter_mut() {
+                                            *pixel = Argb::FULL_GREEN;
+                                            pixel.0 = Some(CONE_LED_STRIP_DIMMING_DEFAULT);
+                                        }
+                                    }
+                                    3 => {
+                                        cone.queue_lcd_fill(Argb::FULL_BLUE)?;
+                                        for pixel in pixels.iter_mut() {
+                                            *pixel = Argb::FULL_BLUE;
+                                            pixel.0 = Some(CONE_LED_STRIP_DIMMING_DEFAULT);
+                                        }
+                                    }
+                                    4 => {
+                                        // show logo if file exists
+                                        let filename = "logo.bmp";
+                                        if std::path::Path::new(filename).exists() {
+                                            cone.queue_lcd_bmp(String::from(filename))?;
+                                        } else {
+                                            tracing::debug!("🚨 File not found: {filename}");
+                                            cone.queue_lcd_fill(Argb::FULL_BLACK)?;
+                                        }
+                                        for pixel in pixels.iter_mut() {
+                                            *pixel = Argb(
+                                                Some(CONE_LED_STRIP_DIMMING_DEFAULT),
+                                                // random
+                                                rand::random::<u8>()
+                                                    % CONE_LED_STRIP_MAXIMUM_BRIGHTNESS,
+                                                rand::random::<u8>()
+                                                    % CONE_LED_STRIP_MAXIMUM_BRIGHTNESS,
+                                                rand::random::<u8>()
+                                                    % CONE_LED_STRIP_MAXIMUM_BRIGHTNESS,
+                                            );
+                                        }
+                                    }
+                                    5 => {
+                                        cone.queue_lcd_qr_code(String::from(
+                                            "https://www.worldcoin.org/",
+                                        ))?;
+                                        for pixel in pixels.iter_mut() {
+                                            *pixel = Argb::DIAMOND_USER_AMBER;
+                                        }
+                                    }
+                                    _ => {}
+                                }
+                                cone.queue_rgb_leds(&pixels)?;
+                            }
+                        } // end tokio::select!
```

Comment:
This is often a sign that the thing inside the {} should be broken into a separate function or closure :)